src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -157,6 +158,45 @@
      */
     List<SimpleInterval> intervalsForTraversal;
 
+    /**
+     * Get the {@link ReferenceDataSource} for this {@link GATKTool}.
+     * Will throw a {@link GATKException} if the reference is null.
+     * Clients are expected to call the {@link #hasReference()} method prior to calling this.
+     * @return the {@link ReferenceDataSource} for this {@link GATKTool}.  Never {@code null}.
+     */
+    protected ReferenceDataSource getReferenceDataSource() {
+        if ( reference == null ) {
+            throw new GATKException("Attempted to retrieve null reference!");
+        }
+        return reference;
+    }
+
+    /**
+     * Get the {@link ReadsDataSource} for this {@link GATKTool}.
```

Comment:
Add disclaimer: "Should only be called by walker base classes in the engine (such as ReadWalker), or by "free-form" tools that extend the GATKTool class directly rather than one of the built-in walker types. Tools that extend a walker type should get their data via apply() rather than directly accessing the engine datasources."

Reply:
Fixed.